spraybot_row_detection/spraybot_row_detection/processing.py

```diff
@@ -49,13 +47,66 @@ def group_points(self, points):
             or len(right_pts) < self.params_['min_pts_row']
         ):
             self.node_.get_logger().info(
-                'Not enough points. Points in left %d Points in right %d'
+                'Not enough points. Points in left_pts %d Points in right_pts %d'
                 % (len(left_pts), len(right_pts))
             )
             return []  # no valid points
         all_pts.extend([left_pts, right_pts])
         return all_pts
 
+    def fit_parallel_lines(self, point_list):
+        # Implementation of parallel lines fitting in https://ieeexplore.ieee.org/document/6697124
+        left_pts = Point.points_to_array(point_list[0])
+        right_pts = Point.points_to_array(point_list[1])
+        A = np.zeros((3, 3))
+        A[0, 0] = np.sum(left_pts[:, 0])          # sum of x in left_pts
+        A[0, 1] = left_pts.shape[0]               # num of left_pts points
+        A[1, 0] = np.sum(right_pts[:, 0])         # sum of x in right_pts
+        A[1, 2] = right_pts.shape[0]              # num of righhjt points
+        A[2, 0] = np.sum(left_pts[:, 0] * left_pts[:, 0]) + np.sum(
+            right_pts[:, 0] * right_pts[:, 0]
+        )                                         # sum of x^2 in left_pts and right_pts
+        A[2, 1] = np.sum(left_pts[:, 0])          # sum of x in left_pts
+        A[2, 2] = np.sum(right_pts[:, 0])         # sum of x in right_pts
+
+        B = np.zeros((3, 1))
+        B[0, 0] = np.sum(left_pts[:, 1])          # sum of y in left_pts
+        B[1, 0] = np.sum(right_pts[:, 1])         # sum of y in right_pts
+        B[2, 0] = np.sum(left_pts[:, 0] * left_pts[:, 1]) + np.sum(
+            right_pts[:, 0] * right_pts[:, 1]
+        )                                         # sum of xy in left_pts and right_pts
```

Comment:
```suggestion
        A[0, 0] = A[2, 1] = np.sum(left_pts[:, 0])          # sum of x in left_pts
        A[1, 0] = A[2, 2] = np.sum(right_pts[:, 0])         # sum of x in right_pts
        A[2, 0] = np.sum(left_pts[:, 0]**2) + np.sum(right_pts[:, 0]**2) # sum of x^2 in left_pts and right_pts
        A[0, 1] = left_pts.shape[0]               # num of left_pts points
        A[1, 2] = right_pts.shape[0]              # num of right points
        )                                         

        B = np.zeros((3, 1))
        B[0, 0] = np.sum(left_pts[:, 1])          # sum of y in left_pts
        B[1, 0] = np.sum(right_pts[:, 1])         # sum of y in right_pts
        B[2, 0] = np.sum(left_pts[:, 0] * left_pts[:, 1]) + np.sum(
            right_pts[:, 0] * right_pts[:, 1]
        )                                         # sum of xy in left_pts and right_pts
```
Simplify. Check for any mathematical inaccuracies.